uaclient/update_info.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict
+
+from uaclient import util
+from uaclient.config import UAConfig
+
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+# All our unit tests are mocking the Cache object, so we are not losing
+# anything here. If by any chance there is no python3-apt in the future, the
+# integration tests should be able to detect it through failure, because we
+# don't expect an empty apt cache there.
+try:
+    from apt import Cache  # type: ignore
+except ImportError:
+
+    def Cache():
+        return []
+
+
+ESM_SERVICES = ("esm-infra", "esm-apps")
+
+series = util.get_platform_info()["series"]
+
+
+def archive_to_service(archive):
+    if archive in (
+        "{}-infra-updates".format(series),
+        "{}-infra-security".format(series),
+    ):
+        return "esm-infra"
+    if archive in (
+        "{}-apps-updates".format(series),
+        "{}-apps-security".format(series),
+    ):
+        return "esm-apps"
+    return archive
+
+
+def get_service_status(status, service_name):
+    for service in status["services"]:
+        if service["name"] == service_name:
+            return (service["entitled"], service["status"])
+
+
+def update_info(cfg: UAConfig) -> Dict[str, Any]:
+    output = {
+        "summary": {
+            "attached": False,
+            "enabled_services": [],
+            "entitled_services": [],
+        },
+        "upgradable_packages": [],
+    }  # type: Dict[str, Any]
+    cache = Cache()
+    status = cfg.status()
+
+    is_attached = status.get("attached")
+    output["summary"]["attached"] = is_attached
+
+    if is_attached:
+        for service in ESM_SERVICES:
+            entitled, enabled = get_service_status(status, service)
+            if entitled == "yes":
+                output["summary"]["entitled_services"].append(service)
+            if enabled == "enabled":
+                output["summary"]["enabled_services"].append(service)
+
```

Comment:
We already have ways of attaining entitled/enabled status that doesn't require interpreting the results of the current schema/values of `cfg.status()`
```suggestion
            ent = entitlements.ENTITLEMENT_CLASS_BY_NAME[service](cfg)
            if ent.contract_status() == ContractStatus.ENTITLED:
                output["summary"]["entitled_services"].append(service)
            if ent.application_status()[0] == status.ApplicationStatus.ENABLED:
                output["summary"]["enabled_services"].append(service)
```

Reply:
nice, will update